scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableUnless.scala

```diff
@@ -0,0 +1,113 @@
+package scalafix.internal.rule
+
+import metaconfig.{Conf, Configured}
+
+import scala.meta._
+import scala.meta.transversers.Traverser
+import scalafix._
+import scalafix.internal.config.DisableUnlessConfig
+import scalafix.internal.util.SymbolOps
+import scalafix.lint.LintCategory
+import scalafix.rule.RuleCtx
+import scalafix.util.SymbolMatcher
+
+object DisableUnless {
+
+  /**
+    * Searches for specific nodes in Tree and adds result of searching to the list.
+    * Also passes context through tree traversal.
+    * fn can either return a value. that stops the searching, or change the context.
+    */
+  class SearcherWithContext[T, U](initContext: U)(
+      fn: PartialFunction[(Tree, U), Either[T, U]])
+      extends Traverser {
+    private var context: U = initContext
+    private val buf = scala.collection.mutable.ListBuffer[T]()
+    // it can be immutable...
+
+    override def apply(tree: Tree): Unit = {
+      if (fn.isDefinedAt((tree, context))) {
```

Comment:
I would use `fn.lift`  or `fn.applyOrElse` to avoid running the pattern twice.

Reply:
`.collect`, `.transform`, `.traverse` methods also use `.isDefinedAt` and `apply`

Reply:
Not anymore https://github.com/scalameta/scalameta/pull/1201

Reply:
nice :D